src/RecipeInstaller.php

```diff
@@ -82,10 +85,11 @@ protected function installProjectFiles(
      * @param string $destinationRoot Base of destination directory (no trailing slash)
      * @param string $sourcePath Full filesystem path to the file to copy
      * @param array $installedFiles List of installed files
-     * @return bool|string
+     * @return array [string, bool]
      */
-    protected function installProjectFile($sourceRoot, $destinationRoot, $sourcePath, $installedFiles)
+    protected function installProjectFile($sourceRoot, $destinationRoot, $sourcePath, $installedFiles): array
```

Comment:
```suggestion
    protected function installProjectFile($sourceRoot, $destinationRoot, $sourcePath, $installedFiles)
```
This would be an API change.  Protected methods are API.

Changing the actual return to an array is also functionally an API change ... so not something we really want to do

Creating a private field `$skippedModules = []` and updating it within `installProjectFile()` might work as an alternative?

Reply:
Reworked in a non-API-breaking way 🧷

Reply:
Great ta. Is there a scenario where this method is going to be called more than once? If so, we'll need to reset the bool back to false after we're done using it

Reply:
Ah yeah I see that if there are `project-files` and `public-files` then it will go twice through the loop. In that case it would write a second time, even if there was nothing else to write. So that's probably enough of a use-case to reset the value.